db/table_cache.cc

```diff
@@ -31,7 +31,23 @@
 #include "util/coding.h"
 #include "util/stop_watch.h"
 
+namespace ROCKSDB_NAMESPACE {
+namespace {
+template <class T>
+static void DeleteEntry(const Slice& /*key*/, void* value) {
+  T* typed_value = reinterpret_cast<T*>(value);
+  delete typed_value;
+}
+}  // namespace
+}  // namespace ROCKSDB_NAMESPACE
+
+// Generate the regular and coroutine versions of some methods by
+// including table_cache_coro.h twice
 // clang-format off
+#define WITHOUT_COROUTINES
+#include "db/table_cache_coro.h"
+#undef WITHOUT_COROUTINES
```

Comment:
Why the` table_cache_coro.h` is added twice? May be adding comment could explain more.

Reply:
The first time, its added with ```#define WITHOUT_COROUTINES```, which will expand the code to a non-coroutine version. The second time its added with ```#define WITH_COROUTINES```, which will expand it to the coroutine version. Basically, you'll get 2 definitions of the function -
```Status TableCache::MultiGet``` and ```folly::coro::Task<Status> TableCache::MultiGetCoroutine```.

I'll add some comments to make it clear.